plugins/mcreator-core/blockly/js/field_mcitem_selector.js

```diff
@@ -53,6 +53,7 @@ class FieldMCItemSelector extends Blockly.FieldImage {
                 let thisField = this; // reference to this field, to use in the callback function
                 javabridge.openMCItemSelector(this.supported_mcitems, {
                     'callback': function (selected) {
+                        Blockly.Events.setGroup(true);

```

Comment:
There are quite some Blockly specifics I don't understand fully (did not dig into it so deep so far), one of them being why this call is now needed?

Reply:
AFAIR, I could not get the new block mutations to get undone/redone properly without creating a new changes group here.

Reply:
>Any source where the need for this is explain.

Not sure to understand the wording.

Reply:
Ok checking some docs call to this starts a chain of events and also needs additional call to end such series of events. If this API is used, it should be used properly and fully

Reply:
> AFAIR, I could not get the new block mutations to get undone/redone properly without creating a new changes group here.

I still don't get this fully. Without specifying this here, did only mutations related to those fields not work, or mutations in general? It is hard to imagine not putting this here making mutations not work at all.

Reply:
> AFAIR, I could not get the new block mutations to get undone/redone properly without creating a new changes group here.

I still don't get this fully. Without specifying this here, did only mutations related to those fields not work, or mutations in general? It is hard to imagine not putting this here making mutations not work at all.

If this is needed app-wide, this should be new PR likely

Reply:
If new blocks had mutations applied, editing an entry selector field (any of 3) would be included in the same group before I added these calls. Does this mean it requires another pull request?

Reply:
We already have blocks with mutations so this problem is probably related to all cases, not probably just this one.